src/WebJobs.Script.Abstractions/Description/IFunctionProvider.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.Abstractions.Description
+{
+    public interface IFunctionProvider
+    {
+        /// <summary>
+        /// Gets any function errors that may occur as part of the provider context
+        /// </summary>
+        /// <returns> An ImmutableDictionary of function name to the array of errors</returns>
+        ImmutableDictionary<string, ImmutableArray<string>> FunctionErrors { get; }
```

Comment:
Why here and below do these have to be ImmutableArrays rather than just IEnumerables? IEnumerable is more natural.

Reply:
hmm, yeah initially I tried to keep this same as `IFunctionMetadataProvider` which uses immutable. I guess just so if it's indexed, it should never ever be changed.
But I don't have a strong preference here.

Changing the contract later would be a breaking change. So yeah good point, we should be certain about the types.

Because it's reconstructed into a compiled list anyway in MetadataManger. I suppose IEnumerable would work fine.